src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -0,0 +1,884 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef FOUR_C_IO_INPUT_SPEC_BUILDERS_HPP
+#define FOUR_C_IO_INPUT_SPEC_BUILDERS_HPP
+
+#include "4C_config.hpp"
+
+#include "4C_io_input_parameter_container.hpp"
+#include "4C_io_input_spec.hpp"
+#include "4C_io_value_parser.hpp"
+
+#include <functional>
+#include <optional>
+#include <ostream>
+#include <variant>
+
+FOUR_C_NAMESPACE_OPEN
+
+namespace Core::IO
+{
+  namespace Internal
+  {
+    /**
+     * Helper to print values in the format of a dat file.
+     */
+    struct DatPrinter
+    {
+      template <typename T>
+      void operator()(std::ostream& out, const T& val) const
+      {
+        out << " " << val;
+      }
+
+      void operator()(std::ostream& out, const std::string& val) const { out << " " << val; }
+
+      void operator()(std::ostream& out, bool val) const { out << " " << std::boolalpha << val; }
```

Comment:
This function has a side effect. Why not

```suggestion
      void operator()(std::ostream& out, bool val) const { out << " " << val ? "true" : "false"; }
```
?